pkg/reporter/reporter.go

```diff
@@ -32,15 +37,47 @@ const (
 
 // Params are the parameters for writing reports.
 type Params struct {
+	HasDiff   bool
+	Details   string
 	Dir       string
-	Operation string
 	IsDestroy bool
-	Output    string
-	HasDiff   bool
+	Message   string
+	Operation string
 }
 
 // Reporter defines the minimum interface for a reporter.
 type Reporter interface {
 	// CreateStatus reports the status of a run.
 	CreateStatus(ctx context.Context, status Status, params *Params) error
+
+	// ClearStatus clears any existing statuses that can be removed.
+	ClearStatus(ctx context.Context) error
+}
+
+type Config struct {
+	GitHub github.Config
+}
+
+func NewReporter(ctx context.Context, t string, c *Config, stdout io.Writer) (Reporter, error) {
```

Comment:
Technically, exported top-level types and functions should be documented 🤷

Reply:
d'oh!